src/pretix/presale/views/order.py

```diff
@@ -197,7 +198,10 @@ def dispatch(self, request, *args, **kwargs):
         return super().dispatch(request, *args, **kwargs)
 
     def post(self, request, *args, **kwargs):
-        resp = self.payment_provider.payment_perform(request, self.order)
+        try:
+            resp = self.payment_provider.payment_perform(request, self.order)
+        except PaymentException as e:
+            return redirect(resp or self.get_order_url() + '?thanks=no')
```

Comment:
With this, we would just ignore the error mesasge instead of showing it to the user, which is not the point. I'd suggest:
```
except PaymentException as e:
    messages.error(request, str(e))
    return redirect(resp or self.get_order_url())
```

The ``thanks=no`` is not required.